src/middlewared/middlewared/plugins/pool.py

```diff
@@ -3382,6 +3388,61 @@ class Config:
         await self.middleware.call('service.restart', 'cron')
         return response
 
+    @accepts(Str('name'), Int('threshold', default=35))
+    async def run(self, name, threshold):
+        """
+        Initiate a scrub of a pool `name` if last scrub was performed more than `threshold` days before.
+        """
+        await self.middleware.call('alert.oneshot_delete', 'ScrubNotStarted', name)
+        await self.middleware.call('alert.oneshot_delete', 'ScrubStarted', name)
+        try:
+            started = await self.__run(name, threshold)
+        except ScrubError as e:
+            await self.middleware.call('alert.oneshot_create', 'ScrubNotStarted', {
+                'pool': name,
+                'text': e.errmsg,
+            })
+        else:
+            if started:
+                await self.middleware.call('alert.oneshot_create', 'ScrubStarted', name)
+
+    async def __run(self, name, threshold):
+        if name == 'freenas-boot':
+            if not await self.middleware.call('system.is_freenas'):
+                if await self.middleware.call('failover.status') == 'BACKUP':
+                    return
+
+            pool = await self.middleware.call('zfs.pool.query', [['name', '=', name]], {'get': True})
+        else:
+            pool = await self.middleware.call('pool.query', [['name', '=', name]], {'get': True})
+            if pool['status'] == 'OFFLINE':
+                if not pool['is_decrypted']:
+                    raise ScrubError(f'Pool {name} is not decrypted, skipping scrub')
+                else:
+                    raise ScrubError(f'Pool {name} is offline, not running scrub')
+
+        if pool['scan']['state'] == 'SCANNING':
+            return False
+
+        history = (await run('zpool', 'history', name, encoding='utf-8')).stdout
+        for match in reversed(list(RE_HISTORY_ZPOOL_SCRUB.finditer(history))):
+            last_scrub = datetime.strptime(match.group(1), '%Y-%m-%d.%H:%M:%S')
+            break
+        else:
+            # creation time of the pool if no scrub was done
+            for match in RE_HISTORY_ZPOOL_CREATE.finditer(history):
+                last_scrub = datetime.strptime(match.group(1), '%Y-%m-%d.%H:%M:%S')
+                break
+            else:
+                raise ScrubError(f'Skipping scrubbing of pool {name}: can\'t get last scrubbing date')
```

Comment:
I am not sure we should raise an error if we cant find the pool create time? Perhaps just log it and let it start the scrub?